packages/web-components/fast-element/src/directives/when.ts

```diff
@@ -1,77 +1,26 @@
-    ExecutionContext,
-    Expression,
-    ObservableExpression,
-} from "../observation/observable";
-import { Behavior } from "./behavior";
-import { Directive } from "./directive";
-
-export class WhenBehavior implements Behavior {
-    private view: SyntheticView | null = null;
-    private cachedView?: SyntheticView;
-    private source: unknown;
-    private observableExpression: ObservableExpression;
-    private context: ExecutionContext | undefined = void 0;
-
-    constructor(
-        private location: Node,
-        expression: Expression,
-        private template: SyntheticViewTemplate
-    ) {
-        this.observableExpression = new ObservableExpression(expression, this);
-    }
-
-    bind(source: unknown, context: ExecutionContext): void {
-        this.source = source;
-        this.context = context;
-        this.updateTarget(this.observableExpression.evaluate(source, context));
-    }
-
-    unbind(): void {
-        if (this.view !== null) {
-            this.view.unbind();
-        }
-
-        this.observableExpression.dispose();
-        this.source = null;
-    }
-
-    handleExpressionChange(): void {
-        this.updateTarget(this.observableExpression.evaluate(this.source, this.context!));
-    }
-
-    updateTarget(show: boolean): void {
-        if (show && this.view == null) {
-            this.view = this.cachedView || (this.cachedView = this.template.create());
-            this.view.bind(this.source, this.context!);
-            this.view.insertBefore(this.location);
-        } else if (!show && this.view !== null) {
-            // do not dispose, since we may want to use the view again
-            this.view.remove();
-            this.view.unbind();
-            this.view = null;
-        }
-    }
-}
-
-export class WhenDirective extends Directive {
-    createPlaceholder: (index: number) => string = DOM.createBlockPlaceholder;
-
-    constructor(public expression: Expression, public template: SyntheticViewTemplate) {
-        super();
-    }
+import { ExecutionContext, Expression } from "../observation/observable";
+
+/**
+ * A directive that enables basic conditional rendering in a template.
+ * @param condition The condition to test for rendering.
+ * @param templateOrTemplateExpression The template or an expression that gets
+ * the template to render when the condition is true.
+ */
+export function when<T = any, K = any>(
+    condition: Expression<T, K>,
+    templateOrTemplateExpression:
+        | SyntheticViewTemplate
+        | Expression<T, SyntheticViewTemplate>
+): CaptureType<T> {
+    let getTemplate: Expression<T, SyntheticViewTemplate>;
 
-    public createBehavior(target: any): WhenBehavior {
-        return new WhenBehavior(target, this.expression, this.template);
+    if (typeof templateOrTemplateExpression === "function") {
+        getTemplate = templateOrTemplateExpression;
+    } else {
+        getTemplate = (): SyntheticViewTemplate => templateOrTemplateExpression;
     }
```

Comment:
nit: i'd suggest:
```typescript
const getTemplate: Expression<T, SyntheticViewTemplate> = typeof templateOrTemplateExpression === "function"
    ? templateOrTemplateExpression
    : (): SyntheticViewTemplate => templateOrTemplateExpression;
```

or:
```typescript
return (source: any, context: ExecutionContext): SyntheticViewTemplate | null =>
    !condition(source, context)
        ? null
        : typeof templateOrTemplateExpression === "function"
        ? templateOrTemplateExpression(source, context)
        : (): SyntheticViewTemplate => templateOrTemplateExpression(source, context);
```

Reply:
I'll refactor to the first one. The second would be a bit slower due to repeated type checks, where as the first option does it one time for all instances created from the template.